examples/language-features/sets.qnt

```diff
@@ -0,0 +1,35 @@
+module sets {
+  var n: int
+  var s: Set[int]
+
+  action Init = all {
+    n' = 0,
+    s' = Set(0)
+  }
+
+  action Next = all {
+    n' = n + 1,
+    s' = s.union(Set(n + 1)),
+    s.exists(x => x == n),
+    s.forall(x => x >= 0),
+    n.in(s),
+    s.contains(n),
+    (n + 1).notin(s),
+    (n + 1).in(union(s, Set(n + 1))),
+    Set(n) == s.intersect(Set(n)),
+    n.notin(s.exclude(Set(n))),
+    Set(n).subseteq(s),
+    Set() == s.filter(x => x > n),
+    s == s.map(x => x),
+    n <= s.fold(0, (x, y) => x + y),
+    s.in(powerset(s)),
+    s == flatten(Set(s)),
+    // TODO: Enable after support for lists is added
+    // Set(List(1)) == Set(1).allLists(),
```

Comment:
`allLists` behaves differently. It is like `Seq(S)` in TLA+, which constructs the (infinite) set of all sequences  over elements from `S`. I am not sure, we actually want `allLists` in Quint.